go-controller/pkg/libovsdbops/portgroup.go

```diff
@@ -75,6 +75,47 @@ func CreateOrUpdatePortGroups(nbClient libovsdbclient.Client, pgs ...*nbdb.PortG
 	return err
 }
 
+// GetPortGroup looks up a port group from the cache
+func GetPortGroup(nbClient libovsdbclient.Client, pg *nbdb.PortGroup) (*nbdb.PortGroup, error) {
+	found := []*nbdb.PortGroup{}
+	opModel := operationModel{
+		Model:          pg,
+		ExistingResult: &found,
+		ErrNotFound:    true,
+		BulkOp:         false,
+	}
+
+	m := newModelClient(nbClient)
+	err := m.Lookup(opModel)
+	if err != nil {
+		return nil, err
+	}
+
+	return found[0], nil
+}
+
+// PortGroupHasPorts returns true if a port group contains all given ports
+func PortGroupHasPorts(nbClient libovsdbclient.Client, pgName string, portUUIDs []string) bool {
+	pg := &nbdb.PortGroup{
+		Name: pgName,
+	}
+	pg, err := GetPortGroup(nbClient, pg)
+	if err != nil {
+		return false
+	}
+loop:
+	for _, portUUID := range portUUIDs {
+		for _, existingPort := range pg.Ports {
+			if portUUID == existingPort {
+				continue loop
+			}
+		}
+		return false
+	}
+
+	return true
```

Comment:
Did you consider using sets?

Reply:
will move to using sets for code consistency